src/test/javascript/spec/component/statistics/statistics.component.spec.ts

```diff
@@ -0,0 +1,70 @@
+import * as chai from 'chai';
+import * as sinonChai from 'sinon-chai';
+import * as sinon from 'sinon';
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { ArtemisTestModule } from '../../test.module';
+import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.service';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { MockComponent, MockDirective, MockPipe } from 'ng-mocks';
+import { AlertComponent } from 'app/shared/alert/alert.component';
+import { RouterTestingModule } from '@angular/router/testing';
+import { MockHasAnyAuthorityDirective } from '../../helpers/mocks/directive/mock-has-any-authority.directive';
+import { TranslatePipe } from '@ngx-translate/core';
+import { ArtemisDatePipe } from 'app/shared/pipes/artemis-date.pipe';
+import { MomentModule } from 'ngx-moment';
+import { StatisticsComponent } from 'app/admin/statistics/statistics.component';
+import { StatisticsGraphComponent } from 'app/admin/statistics/statistics-graph.component';
+import { SpanType } from 'app/entities/statistics.model';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('StatisticsComponent', () => {
+    let fixture: ComponentFixture<StatisticsComponent>;
+    let component: StatisticsComponent;
+
+    beforeEach(fakeAsync(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, RouterTestingModule.withRoutes([]), MomentModule],
+            declarations: [
+                StatisticsComponent,
+                MockComponent(AlertComponent),
+                MockComponent(StatisticsGraphComponent),
+                MockDirective(MockHasAnyAuthorityDirective),
+                MockPipe(TranslatePipe),
+                MockPipe(ArtemisDatePipe),
+            ],
+            providers: [
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(StatisticsComponent);
+                component = fixture.componentInstance;
+            });
+    }));
+
+    afterEach(fakeAsync(() => {
+        jest.clearAllMocks();
```

Comment:
Same here.

Reply:
I use a .restore() call in the end of the test which uses a sinon call. Shouldn't this be enough?

Reply:
It is enough. I was more commenting since I prefer having the restore call in after all. (Makes it easier to add tests in the future if functionality gets extended.)